### DISCUSSION TITLE
React Native - Hermes

### DISCUSSION THREAD

**Discussion by LunatiqueCoder** [reduxjs/redux-devtools#1429] [Q&A] [answered]
Hello!

I really love this tool! And I don't plan to give up on it.

React Native is having some nice updates, especially some debugging improvements, however, our old react-native-debugger which integrates Redux DevTools is getting deprecated and I've been struggling to use Redux DevTools with React Native since then but with no success.

### **Here's what I've tried:** 

**1.** Patch `@redux-devtools/remote` according to https://github.com/reduxjs/redux-devtools/issues/1382

**2.** Use `@redux-devtools/cli` to start a local server by running `redux-devtools --port=8000 --open`

**3.** Setup store

```tsx
let enhancers: StoreEnhancer[] = [];

if (__DEV__) {
  require('react-native-get-random-values');
  const { devToolsEnhancer } = require('@redux-devtools/remote');

  enhancers.push(
    devToolsEnhancer({
      name: Platform.OS,
      hostname: Platform.select({ ios: 'localhost', android: '10.0.2.2' }),
      port: 8000,
      secure: false,
      realtime: true,
    })
  );
}

const store = configureStore({
    reducer: rootReducer,
    enhancers
  });
```

<br>

**4.** Run app

<img width="269" alt="image" src="https://github.com/reduxjs/redux-devtools/assets/55203625/f8492b88-7c64-4e46-9e1e-44e5861c3c12">

<br>
<br>

**5.** Redux DevTools Settings

<img width="615" alt="image" src="https://github.com/reduxjs/redux-devtools/assets/55203625/3d3a5a26-78a6-45d9-90e3-392fdb671830">

### **Results:**

<img width="616" alt="image" src="https://github.com/reduxjs/redux-devtools/assets/55203625/cbe813c5-8710-4d37-ba1d-14c1022c9565">

<br>
<br>
<br>
<br>

Any help is much appreciated. Thank you very much! Cheers

**Comment by LunatiqueCoder** [comment, marked as answer]
Fixed it by patching `@redux-devtools/remote/lib/cjs/devTools.js` like this: https://github.com/reduxjs/redux-devtools/issues/1382#issuecomment-1615995161

**EDIT:**

Also opened a PR here to fix this for React Native world: https://github.com/reduxjs/redux-devtools/pull/1430

**Comment by gemartinez56** [comment]
hi @LunatiqueCoder - thank you for sharing this. Which packages did you have to install to get your server running?
